src/infra/core/src/repos/dataset_impl.rs

```diff
@@ -414,16 +421,13 @@ where
 
         let mut new_upstream_ids: Vec<opendatafabric::DatasetID> = vec![];
         if let opendatafabric::MetadataEvent::SetTransform(transform) = &event {
-            for new_input in transform.inputs.iter() {
-                if let Some(id) = &new_input.id {
+            for new_input in &transform.inputs {
+                if let Some(id) = new_input.dataset_ref.id() {
                     new_upstream_ids.push(id.clone());
                 } else {
-                    return Err(CommitError::MetadataAppendError(AppendError::InvalidBlock(
-                        AppendValidationError::InvalidEvent(InvalidEventError::new(
-                            event,
-                            "Transform input with unresolved ID",
-                        )),
-                    )));
+                    // Normally all references must be resolved to IDs already.
+                    // We continue here while expecting MetadataChain to reject
+                    // this event.
```

Comment:
Previously we had:
```rust
return Err(CommitError::MetadataAppendError(AppendError::InvalidBlock(
    AppendValidationError::InvalidEvent(InvalidEventError::new(
        event,
        "Transform input with unresolved ID",
    )),
)));
```
**Q:** Could you please clarify: how will the event be rejected?